SPECIFICATION_TESTING.md

```diff
@@ -0,0 +1,136 @@
+<!-- TOC -->
```

Comment:
We don't have these markers in other docs, maybe remove? The [Markdown all in one](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one) vscode plugin doesn't require them.

Reply:
It appears in [DEVELOPER_GUIDE](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md?plain=1#L1), too. Shall I remove it there too?

Reply:
Yes, but I am nitpicking.

Reply:
Note that it's not valid markdown ;)

Reply:
It's just an HTML comment, and inline raw HTML is valid according to:
- [GitHub Flavored Markdown](https://github.github.com/gfm/#raw-html)
- [CommonMark](https://spec.commonmark.org/0.31.2/#raw-html)
- [John Gruber's Original Definition](https://daringfireball.net/projects/markdown/syntax#html)

Reply:
I stay corrected.